frontend/src/app/containers/App.js

```diff
@@ -93,21 +98,72 @@ class App extends Component {
       locale: (navigator.language || '').split('-')[0]
     });
     this.props.chooseTests();
-    this.props.fetchUserCounts(config.usageCountsURL).then(() => {
+    const userCountsPromise = this.props.fetchUserCounts(config.usageCountsURL);
+    this.measurePageview();
+
+    const langs = {};
+
+    function addLang(lang, response) {
+      if (response.ok) {
+        return response.text().then(data => {
+          langs[lang] = `${langs[lang] || ''}${data}
+`;
+        });
+      }
+      return Promise.resolve();
+    }
+
+    const negotiated = negotiateLanguages(
+      navigator.languages,
+      config.AVAILABLE_LOCALES,
+      { defaultLocale: 'en-US' }
+    );
+
+    const promises = negotiated.map(language =>
+      Promise.all(
+        [
+          fetch(`/static/locales/${language}/app.ftl`).then(response => addLang(language, response)),
```

Comment:
Yeah, this bit where the .ftl resources are loaded... #2229 could be fixable by fetching another resource first, say `languages.json` that has contents like this generated in gulp:
```json
{
"en-US": ["/static/locales/en-US/app.08675309.ftl", "/static/locales/en-US/experiments.decafbad.ftl"],
"fr":  ["/static/locales/fr/app.cafebabe.ftl", "/static/locales/en-US/experiments.deadf00d.ftl"]
}
```
Then, do a fetch on those resources. Maybe a crazy idea.